contracts/introspection/ERC165Checker.sol

```diff
@@ -0,0 +1,111 @@
+pragma solidity ^0.4.24;
+
+
+/**
+ * @title ERC165Checker
+ * @dev Use `using ERC165Checker for address`; to include this library
+ *      https://github.com/ethereum/EIPs/blob/master/EIPS/eip-165.md
+ */
+library ERC165Checker {
+  // As per the EIP-165 spec, no interface should ever match 0xffffffff
+  bytes4 private constant InterfaceId_Invalid = 0xffffffff;
+
+  bytes4 private constant InterfaceId_ERC165 = 0x01ffc9a7;
+  /**
+   * 0x01ffc9a7 ===
+   *   bytes4(keccak256('supportsInterface(bytes4)'))
+   */
+
+  /**
+   * @notice Query if a contract implements an interface, does not check ERC165 support
+   * @param _address The address of the contract to query for support of an interface
+   * @param _interfaceId The interface identifier, as specified in ERC-165
+   * @return true if the contract at _address indicates support of the interface with
+   * identifier _interfaceId, false otherwise
+   * @dev Assumes that _address contains a contract that supports ERC165, otherwise
+   *      the behavior of this method is undefined. This precondition can be checked 
+   *      with the `supportsERC165` method in this library. 
+   *      Interface identification is specified in ERC-165.
+   */
+  function supportsERC165Interface(address _address, bytes4 _interfaceId)
+    internal
```

Comment:
This function should be `private` IMO, because it should never be used before also checking `supportsERC165`.

Reply:
I think I like your `supportInterfaces` API more than what we have now. Happy to implement this, lets wait for feedback from others.

Reply:
Me too! I think that makes much more sense, I wasn't too happy with the naming scheme we'd come up with.